JASP-Desktop/analysis/analysisform.cpp

```diff
@@ -267,40 +267,43 @@ void AnalysisForm::_setUpRelatedModels()
 void AnalysisForm::_setUpItems()
 {
 	QList<JASPControlWrapper*> controls = _controls.values();
+
 	for (JASPControlWrapper* control : controls)
 		control->setUp();
 
 	// set the order of the BoundItems according to their dependencies (for binding purpose)
 	for (JASPControlWrapper* control : controls)
 	{
-		QVector<JASPControlWrapper*> depends = control->depends();
-		int index = 0;
-		while (index < depends.length())
+		std::vector<JASPControlWrapper*> depends(control->depends().begin(), control->depends().end());
+
+		// This looks like a for-each loop right? Except that we add a dependency to control.depends and also to our local copy...
+		// Which means that won't work because then we invalidate the loop, instead we do it old-school with an index.
+		// But I thought it would be nice to add a comment here describing the magic going on here for the next forlorn soul looking into this.
+		for (size_t index = 0; index < depends.size(); index++)
 		{
-			JASPControlWrapper* depend = depends[index];
-			const QVector<JASPControlWrapper*>& dependdepends = depend->depends();
+			JASPControlWrapper					* depend		= depends[index];
+			const std::set<JASPControlWrapper*>	& dependdepends = depend->depends();
+
 			for (JASPControlWrapper* dependdepend : dependdepends)
-			{
 				if (dependdepend == control)
 					addFormError(tq("Circular dependency between control ") + control->name() + tq(" and ") + depend->name());
-				else
-				{
-					if (control->addDependency(dependdepend))
-						depends.push_back(dependdepend);
-				}
-			}
-			index++;
+				else if (control->addDependency(dependdepend))
+					depends.push_back(dependdepend);
 		}
 	}
+
 	std::sort(controls.begin(), controls.end(), 
-			  [](JASPControlWrapper* a, JASPControlWrapper* b) {
-					return a->depends().length() < b->depends().length(); 
-			});
+		[](JASPControlWrapper* a, JASPControlWrapper* b) {
+			return a->depends().size() < b->depends().size();
+		});
 
 	for (JASPControlWrapper* control : controls)
 	{
-		_orderedControls.push_back(control);
+		_dependsOrderedCtrls.push_back(control);
+		connect(control->item(), &JASPControlBase::helpMDChanged, this, &AnalysisForm::helpMDChanged);
```

Comment:
Because here the items are setup? So I figured this was a good place?
Where exactly would you suggest is better?

 I had to do quite some guessing where to put places and I was also misdirected for a while because of the original name of `_dependsOrderedCtrls` as I thought it was all `ordered ctrls` but they weren't actually ordered as  I expected :p

It seems to work now so I thought this was fine but Im all ear for any suggestion ^^